src/core/api.c

```diff
@@ -354,6 +354,105 @@ MsQuicConnectionStart(
     return Status;
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
```

Comment:
I don't see any reason why this shouldn't be allowable at DISPATCH_LEVEL. You just queue and return.